scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -2,18 +2,90 @@ package scalafix
 package internal.config
 
 import metaconfig._
+import metaconfig.annotation.{Description, ExampleValue}
+import metaconfig.generic.Surface
 import org.langmeta.Symbol
 
 import scalafix.internal.util.SymbolOps
-import metaconfig.annotation.{Description, ExampleValue}
-import metaconfig.generic.Surface
+
+case class DisabledSymbol(
+    @ExampleValue("scala.sys.process.Process")
+    @Description(
+      "A single symbol to ban. Cannot be used together with the regex option.")
+    symbol: Option[Symbol.Global],
+    @Description("Custom message.")
+    message: Option[String],
+    @Description("Custom id for error messages.")
+    id: Option[String],
+    @Description(
+      "Regex to ban symbols matching a given include/exclude patterns." +
+        "Cannot be used together with the symbol option." +
+        "Include and exclude filters can be either a list of regex strings or a single regex string.")
+    @ExampleValue("""
+                    |{
+                    |  includes = [
+                    |    "java.io.*"
+                    |    "scala.io.*"
+                    |  ]
+                    |  excludes = "java.io.InputStream"
+                    |}""".stripMargin)
+    regex: Option[FilterMatcher]) {
+
+  def matches(symbol: Symbol): Boolean = {
+    this.symbol match {
+      case Some(s) => SymbolOps.isSameNormalized(symbol, s)
+      case None =>
+        regex match {
+          case Some(r) => r.matches(symbol.toString)
+          case None => sys.error("impossible")
+        }
+    }
+  }
+}
+
+object DisabledSymbol {
+  implicit val surface: Surface[DisabledSymbol] =
+    generic.deriveSurface[DisabledSymbol]
+
+  private def normalizeMessage(msg: String): String =
+    if (msg.isMultiline) {
+      "\n" + msg.stripMargin
+    } else {
+      msg
+    }
+  implicit val reader: ConfDecoder[DisabledSymbol] =
+    ConfDecoder.instanceF[DisabledSymbol] {
+      case c: Conf.Obj =>
+        (c.getOption[Symbol.Global]("symbol") |@|
+          c.getOption[String]("message") |@|
+          c.getOption[String]("id") |@|
+          c.getOption[FilterMatcher]("regex"))
+          .andThen {
+            case (((Some(_), b), c), Some(_)) =>
+              Configured.notOk(ConfError.message(
+                "Cannot specify both symbol and regex, only one of them is allowed."))
+            case (((a @ Some(_), b), c), None) =>
+              Configured.ok(DisabledSymbol(a, b.map(normalizeMessage), c, None))
+            case (((None, b), c), d @ Some(_)) =>
+              Configured.ok(DisabledSymbol(None, b.map(normalizeMessage), c, d))
+            case (((None, b), c), None) =>
+              Configured.notOk(
+                ConfError.message("Either symbol or regex must be specified."))
+          }
+      case s: Conf.Str =>
+        symbolGlobalReader
+          .read(s)
+          .map(sym => DisabledSymbol(Some(sym), None, None, None))
+      case _ => Configured.NotOk(ConfError.message("Wrong config format"))
```

Comment:
`.instance` allows you to pass a partial function and metaconfig produces a slightly more helpful error message